src/mds/PurgeQueue.h

```diff
@@ -45,11 +50,35 @@ class PurgeItem
   fragtree_t fragtree;
 
   PurgeItem()
-   : action(NONE), ino(0), size(0)
+   : stamp(ceph_clock_now()), pad_size(0), action(NONE), ino(0), size(0)
   {}
 
   void encode(bufferlist &bl) const;
   void decode(bufferlist::const_iterator &p);
+
+  static Action str_to_type(std::string_view str) {
+    return PurgeItem::actions.at(std::string(str));
```

Comment:
Small nit: see for how to lookup using string_view: https://stackoverflow.com/questions/35525777/use-of-string-view-for-map-lookup